src/components/Tokens/TokenTable/TokenTable.tsx

```diff
@@ -149,14 +167,94 @@ export function LoadingTokenTable() {
+  const [topTokens, setTopTokens] = useState<TokenData[]>(
+    Array.from({ length: MAX_TOKENS_TO_LOAD }).map((_) => ({ loading: true } as TokenData))
+  )
+  const [error, setError] = useState<any>(undefined)
+  // const filteredTokens = useFilteredTokens(data)
+  // const sortedFilteredTokens = useSortedTokens(filteredTokens)
+
+  const isItemLoaded = (index: number) => index < topTokens.length && topTokens[index].loading === false
+
+  const loadMoreItems = (startIndex: number, stopIndex: number) => {
+    if (stopIndex >= page * PAGE_SIZE && page < MAX_PAGE) setPage(page + 1)
+    return fetchQuery<TopTokenQueryType>(environment, query, {
+      pageSize: PAGE_SIZE,
+      page,
+    })
+      .toPromise()
+      .then((data) => {
+        const newPageTopTokens: TokenData[] = !!data?.topTokenProjects
+          ? data.topTokenProjects.map((token) =>
+              token?.tokens?.[0].address
+                ? {
+                    loading: false,
+                    name: token?.name,
+                    address: token?.tokens?.[0].address,
+                    chain: token?.tokens?.[0].chain,
+                    symbol: token?.tokens?.[0].symbol,
+                    price: token?.markets?.[0]?.price,
+                    marketCap: token?.markets?.[0]?.marketCap,
+                    volume: {
+                      [TimePeriod.HOUR]: token?.markets?.[0]?.volume1H,
+                      [TimePeriod.DAY]: token?.markets?.[0]?.volume1D,
+                      [TimePeriod.WEEK]: token?.markets?.[0]?.volume1W,
+                      [TimePeriod.MONTH]: token?.markets?.[0]?.volume1M,
+                      [TimePeriod.YEAR]: token?.markets?.[0]?.volume1Y,
+                      [TimePeriod.ALL]: token?.markets?.[0]?.volumeAll,
+                    },
+                    percentChange: {
+                      [TimePeriod.HOUR]: token?.markets?.[0]?.pricePercentChange1H,
+                      [TimePeriod.DAY]: token?.markets?.[0]?.pricePercentChange24h,
+                      [TimePeriod.WEEK]: token?.markets?.[0]?.pricePercentChange1W,
+                      [TimePeriod.MONTH]: token?.markets?.[0]?.pricePercentChange1M,
+                      [TimePeriod.YEAR]: token?.markets?.[0]?.pricePercentChange1Y,
+                      [TimePeriod.ALL]: token?.markets?.[0]?.pricePercentChangeAll,
+                    },
+                  }
+                : ({} as TokenData)
+            )
+          : []
+        const currentTopTokens = [...topTokens]
+        newPageTopTokens.forEach((token, i) => {
+          if (currentTopTokens[startIndex + i].loading) currentTopTokens[startIndex + i] = token
+        })
+        setTopTokens(currentTopTokens)
+      })
+      .catch((e) => setError(e))
+  }
+
+  const Row = function TokenRow({ data, index, style }: TokenRowProps) {
+    const tokenData = data[index]
+    if (!tokenData || topTokens[index].loading) {
+      return <LoadingRow style={style} key={index} />
```

Comment:
is this also a dupe of the loading rows above?

Reply:
no this is different. the above loading table is for the suspense, if we dont have that we see the tiny little round spinner again.